src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -972,7 +972,7 @@ public override ForEachStatementInfo GetForEachStatementInfo(CommonForEachStatem
                 enumeratorInfoOpt.MoveNextMethod.GetPublicSymbol(),
                 currentProperty: ((PropertySymbol)enumeratorInfoOpt.CurrentPropertyGetter?.AssociatedSymbol).GetPublicSymbol(),
                 disposeMethod.GetPublicSymbol(),
-                enumeratorInfoOpt.ElementType.GetPublicSymbol(),
+                enumeratorInfoOpt.ElementTypeWithAnnotations.GetPublicSymbol(),
```

Comment:
I'd like to see a compiler-level test specifically for this change. The best file would probably be `NullablePublicAPITests.cs`

Reply:
Added test. I wasn't sure if it made sense to add to the `GetDeclaredSymbol_Foreach_*` tests so I just made a small new one